clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -126,6 +141,12 @@ public synchronized void addTopics(String... topics) {
         requestUpdate();
     }
 
+    public synchronized void setTopics(Collection<String> topics) {
```

Comment:
This confused me a bit since previously this class was purely additive wrt the set of topics. Is there a case where you can actually lose a subscription by doing this? I'm thinking if the user does something to the subscriptions during a callback?